tools/coverage/design.md

```diff
@@ -0,0 +1,37 @@
+# Design of Test Coverage Tool
+
+See [design.svg](https://github.com/knative/test-infra/tree/master/tools/coverage/design.svg) for the design diagram.
+
+We pack the test coverage tool in a container, that is triggered by prow. It runs test coverage profiling on target repository. Prow clones the target repository as the current working directory for the container.
+
+ Prow has handlers for different github events. We add a pre-submit prow job that is triggered by any new commit to a PR to run test coverage on the new build to compare it with the master branch and previous commit for pre-submit coverage. We add a post-submit prow job that is triggered by merge events, to run test coverage on the nodes of master branch. Test coverage data on the master branch is supplied to TestGrid for displaying the coverage change over time, as well as serve as the basis of comparison for pre-submit coverage mentioned in the pre-submit scenario.
+
+ Here is the step-by-step description of the pre-submit and post-submit workflows
+
+## Pre-submit workflow
+
+Runs code coverage tool to report coverage change in a new commit or new PR
+
+1. Developer submit new commit to an open PR on github
+1. Matching pre-submit prow job is started
+1. Test coverage profile generated
+1. Calculate coverage change against master branch. Compare the coverage file generated in this cycle against the most recent successful post-submit build. Coverage file for post-submit commits were generated in post-submit workflow
```

Comment:
Use the concept you explained in the summary above, instead of "post-submit XYZ".